### PR TITLE
Do not create decision issues if appeal lacks completed BVA Dispatch task

### PR DESCRIPTION
**Pull request by pkarman**
connects #10051 

### Description

Appeals require an additional task be completed before being closed as "decided".

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -673,6 +673,10 @@ def available_review_options
     %w[supplemental_claim cavc]
   end
 
+  def completed_bva_dispatch_task?
+    !!tasks.find { |t| t.is_a?(BvaDispatchTask) }&.completed?
```
[the remark this answers is not in the document]
Comment (aroltsch) [reply app/models/appeal.rb]: I think we should only sync when all BvaDispatchTask tasks are completed. An appeal usually has two BvaDispatchTask tasks: one assigned to user and one assigned to the organization.

File: app/models/request_issue.rb [hunk 2]
```diff
@@ -212,7 +212,7 @@ def end_product_code
   end
 
   def status_active?
-    return appeal_active? if decision_review.is_a?(Appeal)
+    return appeal_active? if decision_review_is_appeal?
```
Comment (leikkisa) [inline app/models/request_issue.rb]: Not related to this PR, but this doesn't seem to handle noncomp claim reviews.

File: app/models/request_issue.rb [hunk 3]
```diff
@@ -440,7 +440,7 @@ def diagnostic_code
 
   def api_status_active?
     return decision_review.active_status? if decision_review.is_a?(ClaimReview)
-    return true if decision_review.is_a?(Appeal)
+    return true if decision_review_is_appeal?
```
Comment (leikkisa) [inline app/models/request_issue.rb]: Again, not really about your PR, but I don't think the api_status_active should just return true for an Appeal. I think that might be a remnant from Melissa not knowing how to check if an appeal was active.

File: app/models/request_issue.rb [hunk 4]
```diff
@@ -534,7 +534,17 @@ def duplicate_of_issue_in_active_review?
...
+  def decision_review_is_appeal?
+    decision_review.is_a?(Appeal)
+  end
+
+  def decision_review_is_appeal_and_does_not_have_completed_bva_dispatch_task?
+    decision_review_is_appeal? && !decision_review.completed_bva_dispatch_task?
+  end
+
   def create_decision_issues
+    return false if decision_review_is_appeal_and_does_not_have_completed_bva_dispatch_task?
```
Comment (leikkisa) [inline app/models/request_issue.rb]: Haha. Just a nit but I think the level of abstraction for the two new methods above might not be necessary here (although it is funny that the method name is about as long as the code itself).

**Comment by pkarman** [comment]
Turns out this case was caused by #9556 and would not be arrived at during any normal business flow.

**Comment by leikkisa** [comment]
I have to admit, I don't understand enough. It looks like the BvaDispatchTask gets created for quality review and dispatch/outcoding, but I'm not sure if it gets created for every appeal.  I wonder if someone in Queue would know.  I also don't know if it matters, because maybe noncomp decision issues get created in a different flow.

I think asking Queue is a good step, but if we don't know it might be better to check that the BvaDispatchTask exists before checking that it's completed.

**Pull request closed without merging**